src/NLog/MappedDiagnosticsLogicalContext.cs

```diff
@@ -175,7 +196,24 @@ public static void Set(string item, string value)
         /// <param name="value">Item value.</param>
         public static void Set(string item, object value)
         {
+            Set<object>(item, value);
+        }
+
+        /// <summary>
+        /// Sets the current logical context item to the specified value.
+        /// </summary>
+        /// <param name="item">Item name.</param>
+        /// <param name="value">Item value.</param>
+        public static void Set<T>(string item, T value)
+        {
+#if NET4_6 || NETSTANDARD
             GetLogicalThreadDictionary(true)[item] = value;
+#else
+            if (Convert.GetTypeCode(value) != TypeCode.Object || typeof(T).IsValueType)
+                GetLogicalThreadDictionary(true)[item] = value;
+            else
+                GetLogicalThreadDictionary(true)[item] = new System.Runtime.Remoting.ObjectHandle(value);
```

Comment:
please do  deduplicate the `GetLogicalThreadDictionary(true)[item]` set, e.g.

```c#
var val = GetSetValue(...)

GetLogicalThreadDictionary(true)[item] = val;
```

Reply:
Fixed